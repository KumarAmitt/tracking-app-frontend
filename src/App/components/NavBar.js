import React from 'react';
import { Link } from 'react-router-dom';
import './style/NavBar.css';

const NavBar = () => (
  <div className="navbar">
    <Link to="/add_deal">Add Deal</Link>
    <Link to="/track">Track</Link>
    <Link to="/progress">Progress</Link>
    <Link to="/profile">Profile</Link>
  </div>
);

export default NavBar;
