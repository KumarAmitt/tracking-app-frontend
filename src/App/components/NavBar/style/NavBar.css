.navbar {
  background-color: var(--accent-secondary);
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  display: flex;
  justify-content: space-between;
}

.navbar > * {
  width: 100%;
  height: 100%;
  padding: 0.6em;
  transition: all linear 300ms;
}

.navbar > *:hover {
  background-color: var(--accent-primary);
}

.navbar a {
  color: #eee;
  text-decoration: none;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.navbar a div {
  font-size: 0.8rem;
  padding: 0.2em;
}
