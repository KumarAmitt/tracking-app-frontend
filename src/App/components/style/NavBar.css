.navbar {
  background-color: var(--accent-secondary);
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  display: flex;
  justify-content: space-between;
  padding: 0.5em;
}

.navbar a {
  color: #eee;
  text-decoration: none;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.navbar a div {
  font-size: 0.8rem;
  padding: 0.2em;
}
