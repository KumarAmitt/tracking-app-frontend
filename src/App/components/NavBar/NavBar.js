import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import AddOutlinedIcon from '@material-ui/icons/AddOutlined';
import TrendingUpOutlinedIcon from '@material-ui/icons/TrendingUpOutlined';
import TrackChangesOutlinedIcon from '@material-ui/icons/TrackChangesOutlined';
import PersonOutlinedIcon from '@material-ui/icons/PersonOutlined';
import './style/NavBar.css';
import { useDispatch, useSelector } from 'react-redux';
import { getUserInfo, loadSession } from '../../../store/slicers/user';

const NavBar = () => {
  const dispatch = useDispatch();
  const userInfo = useSelector(getUserInfo);

  useEffect(() => {
    dispatch(loadSession());
  }, []);

  return (
    <>
      {
          userInfo.logged_in
          && (
          <div className="navbar">
            <Link to="/add_deal">
              <AddOutlinedIcon />
              <div>Add Deal</div>
            </Link>
            <Link to="/track">
              <TrendingUpOutlinedIcon />
              <div>Track</div>
            </Link>
            <Link to="/progress">
              <TrackChangesOutlinedIcon />
              <div>Progress</div>
            </Link>
            <Link to="/profile">
              <PersonOutlinedIcon />
              <div>Profile</div>
            </Link>
          </div>
          )
        }
    </>
  );
};

export default NavBar;
