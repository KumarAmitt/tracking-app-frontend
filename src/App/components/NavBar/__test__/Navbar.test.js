import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureAppStore from '../../../../store/configureStore';
import NavBar from '../NavBar';

const store = configureAppStore();

describe('Navbar', () => {
  it('should match the snapshot of Navbar component', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>
      </Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
