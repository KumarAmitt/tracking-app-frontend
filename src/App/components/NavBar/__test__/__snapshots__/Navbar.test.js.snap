// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should match the snapshot of Navbar component 1`] = `
<div
  className="navbar"
>
  <a
    href="/add_deal"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
    </svg>
    <div>
      Add Deal
    </div>
  </a>
  <a
    href="/track"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M16 6l2.29 2.29-4.88 4.88-4-4L2 16.59 3.41 18l6-6 4 4 6.3-6.29L22 12V6h-6z"
      />
    </svg>
    <div>
      Track
    </div>
  </a>
  <a
    href="/progress"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19.07 4.93l-1.41 1.41C19.1 7.79 20 9.79 20 12c0 4.42-3.58 8-8 8s-8-3.58-8-8c0-4.08 3.05-7.44 7-7.93v2.02C8.16 6.57 6 9.03 6 12c0 3.31 2.69 6 6 6s6-2.69 6-6c0-1.66-.67-3.16-1.76-4.24l-1.41 1.41C15.55 9.9 16 10.9 16 12c0 2.21-1.79 4-4 4s-4-1.79-4-4c0-1.86 1.28-3.41 3-3.86v2.14c-.6.35-1 .98-1 1.72 0 1.1.9 2 2 2s2-.9 2-2c0-.74-.4-1.38-1-1.72V2h-1C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10c0-2.76-1.12-5.26-2.93-7.07z"
      />
    </svg>
    <div>
      Progress
    </div>
  </a>
  <a
    href="/profile"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 6c1.1 0 2 .9 2 2s-.9 2-2 2-2-.9-2-2 .9-2 2-2m0 10c2.7 0 5.8 1.29 6 2H6c.23-.72 3.31-2 6-2m0-12C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 10c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
      />
    </svg>
    <div>
      Profile
    </div>
  </a>
</div>
`;
